deprecation.py

```diff
@@ -183,14 +195,26 @@ def _function_wrapper(function):
             # a number of ways the deprecation notice could go. The following
             # makes for a nicely constructed sentence with or without any
             # of the parts.
-            parts = {
-                "deprecated_in":
-                    " %s" % deprecated_in if deprecated_in else "",
-                "removed_in":
-                    "\n   This will be removed in %s." %
-                    removed_in if removed_in else "",
-                "details":
-                    " %s" % details if details else ""}
+            if isinstance(removed_in, date):
```

Comment:
To avoid repeated code, the branch should only contain the part which varies - minimally, just the preposition. :-)

Reply:
I just realized. I made the change